numpy/core/tests/test_multiarray.py

```diff
@@ -4333,7 +4333,41 @@ def test_object_argmax_with_NULLs(self):
         assert_equal(a.argmax(), 3)
         a[1] = 30
         assert_equal(a.argmax(), 1)
+    
+    def test_np_argmax_keepdims(self):
 
+        sizes = [(3,), (3, 2), (2, 3),
+                 (3, 3), (2, 3, 4), (4, 3, 2),
+                 (1, 2, 3, 4), (2, 3, 4, 1),
+                 (3, 4, 1, 2), (4, 1, 2, 3)]
+        for size in sizes:
+            arr = np.random.normal(size=size)
+            for axis in range(-len(size), len(size)):
+                res_orig = np.argmax(arr, axis=axis)
+                new_shape = list(size)
+                new_shape[axis] = 1
+                res_orig = res_orig.reshape(new_shape)
+                res = np.argmax(arr, axis=axis, keepdims=True)
+                assert_equal(res, res_orig)
+                assert_(res.ndim == arr.ndim)
+                assert_(res.shape[axis] == 1)
+                outarray = np.empty(res.shape, dtype=res.dtype)
```

Comment:
It would be nice to have a few tests with keepdims and `out` that is not already c-contiguous and `arr` that is not contiguous.  Could be here or elsewhere.

It would also be nice if you can check that we have simple error tests for `out` being misshaped (we may already ahve, but should have some with `keepdims=True`).